quantus/helpers/utils.py

```diff
@@ -343,21 +376,41 @@ def get_nr_patches(
     return np.prod(shape) // np.prod(patch_size)
 
 
-def _pad_array(arr: np.array, pad_width: int, mode: str, omit_first_axis=True):
+def pad_array(arr: np.ndarray, pad_width: int, mode: str,
+              omit_first_axis: bool = True, batched: bool = True,
+) -> np.ndarray:
     """To allow for any patch_size we add padding to the array."""
     pad_width_list = [(pad_width, pad_width)] * arr.ndim
-    if omit_first_axis:
+    
+    if batched:
```

Comment:
the batched boolean (everywhere that it is employed) feels a bit weird.

I think a lot less cluttered solution would be to simply always introduce a batch axis during preprocessing if it is not yet present, and then work based on that in all functions.